transaction/transaction/src/main/java/io/helidon/transaction/Tx.java

```diff
@@ -0,0 +1,303 @@
+         * If called inside a transaction context, the current transaction context must be suspended. The method execution
+         * must then continue outside a transaction context, and the previously suspended transaction must be resumed
+         * by the interceptor that suspended it after the method execution has completed.
+         */
+        UNSUPPORTED
+    }
+
+    /**
+     * {@link Type} annotation.
+     * Defines transaction type of the annotated method.
+     */
+    @Target(ElementType.ANNOTATION_TYPE)
+    public @interface TransactionType {
+        /**
+         * Transaction type of the annotated method.
+         *
+         * @return the transaction type
+         */
+        Type value();
+    }
+
+    /**
+     * Defines {@link Type#MANDATORY} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.MANDATORY)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Mandatory { }
+
+    /**
+     * Defines {@link Type#NEW} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.NEW)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface New { }
+
+    /**
+     * Defines {@link Type#NEVER} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.NEVER)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Never { }
+
+    /**
+     * Defines {@link Type#REQUIRED} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.REQUIRED)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Required { }
+
+    /**
+     * Defines {@link Type#SUPPORTED} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.SUPPORTED)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Supported { }
+
+    /**
+     * Defines {@link Type#UNSUPPORTED} transaction type of the annotated method.
+     */
+    @Interception.Intercepted
+    @TransactionType(Type.UNSUPPORTED)
+    @Retention(RetentionPolicy.SOURCE)
+    @Target({ElementType.TYPE, ElementType.METHOD})
+    public @interface Unsupported { }
+
+    /**
+     * Execute provided task as database transaction.
+     * Transaction is handled automatically. Task computes and returns result.
+     *
+     * @param task task to run in transaction
+     * @param <T>  the result type of the task
+     * @return computed task result
+     * @throws TxException when result computation failed
+     */
+    public static <T> T transaction(Callable<T> task) {
```

Comment:
It is very odd that this is called, simply, `transaction`. If I am reading right, it "runs" or "executes" or "calls" the supplied `Callable`, so I would expect a verb here.